### PR TITLE
[GuiQt] Remove some unactivated code

### PR DESCRIPTION
**Pull request by epernod**
Just some cleaning while debugging the gui.

All code are under #ifdef never define.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp [hunk 1]
```diff
@@ -966,17 +715,10 @@ void RealGUI::popupOpenFileSelector()
                                   );
     if ( s.length() >0 )
     {
-#ifdef SOFA_PML
-        if ( s.endsWith ( ".pml" ) )
-            pmlOpen ( s );
-        else if ( s.endsWith ( ".lml" ) )
-            lmlOpen ( s );
+        if (s.endsWith( ".simu") )
+            fileOpenSimu(s.toStdString());
```
Comment (fredroy) [inline Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp:719]: TIL I learn about the `.simu` extension 🙃

**Review by fredroy — CHANGES_REQUESTED** [review]
I think you can also remove all the code related to `SOFA_GUI_QT_NO_RECORDER` as it is always defined (and we have a nice double negation 🤐 `#ifndef SOFA_GUI_QT_NO_RECORDER`)

File: Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp (hunk 1 above)
Comment (epernod) [reply Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp:719]: will remove SOFA_GUI_QT_NO_RECORDER next time I dig into the gui ^^

**Review by alxbilger — COMMENTED** [review]
What was `SOFA_GUI_INTERACTION` supposed to do?

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
> What was `SOFA_GUI_INTERACTION` supposed to do?

I guess nobody knows 🔮

**Comment by epernod** [comment]
> What was `SOFA_GUI_INTERACTION` supposed to do?

I dig into the git history before removing the macro but I was not able to find what it was doing and why the macro is not anymore here (introduce/remove before moving on CMake).

**Pull request merged**